client/extensions/woocommerce/state/ui/payments/methods/test/reducer.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import reducer from '../reducer';
+import {
+	openPaymentMethodForEdit,
+	closeEditingPaymentMethod,
+	cancelEditingPaymentMethod,
+	changePaymentMethodField,
+} from '../actions';
+
+const siteId = 123;
+
+const initialState = {
+	creates: [],
+	updates: [],
+	deletes: [],
+	currentlyEditingId: null,
+};
+
+describe( 'reducer', () => {
+	describe( 'openPaymentMethodForEdit', () => {
+		it( 'should set currentlyEditingId when the method has a server-side ID', () => {
+			const newState = reducer( initialState, openPaymentMethodForEdit( siteId, 1 ) );
```

Comment:
I've been thinking about this sort of thing as well, and I'm not sure that it's really buying a whole lot to hard code actions unless the action creator is very intricate. And the cost we pay is tighter coupling to the action object. i.e. In the case where the action creator is very simple (like this one), we gain more resilient tests by doing it this way. And the only thing we lose is testing a single line return function, which isn't providing much value for us anyway. So in the future, we could change the action structure and as long as we update the action creator and the reducer, this test will still run. (Although we may need to change some `expect` statements depending on the change)